quic/handshake/FizzBridge.h

```diff
@@ -9,7 +9,57 @@
 #pragma once
 
 #include <fizz/crypto/aead/Aead.h>
+#include <quic/handshake/Aead.h>
+
+#include <memory>
+#include <utility>
 
 namespace quic {
-using Aead = fizz::Aead;
-}
+
+class FizzAead final : public Aead {
+private:
+  std::unique_ptr<fizz::Aead> fizzAead;
+  FizzAead(std::unique_ptr<fizz::Aead> _fizzAead)
+      : fizzAead(std::move(_fizzAead)) {}
+
+public:
+  static std::unique_ptr<FizzAead> wrap(std::unique_ptr<fizz::Aead> _fizzAead) {
+    if (!_fizzAead) {
+      return {};
+    }
+
+    return std::unique_ptr<FizzAead>(new FizzAead(std::move(_fizzAead)));
+  }
+
+  /**
+   * Simply forward all calls to fizz::Aead.
+   */
+  std::unique_ptr<folly::IOBuf>
+  encrypt(std::unique_ptr<folly::IOBuf> &&plaintext,
+          const folly::IOBuf *associatedData, uint64_t seqNum) const override {
+    return fizzAead->encrypt(
```

Comment:
since one can wrap a null fizz::aead in, fizzAead can possibly be nullptr. Should you check against that before deref it?

Reply:
The wrap function ensures that this isn't the case.

Reply:
Why? is you pass nullptr to wrap, you get a FizzAead object with the fizzAead member being nullptr, right?

Reply:
The wrap funtion check the pointer that is passed in and return a null unique pointer when that's the case. The constructor is private, so by construct, the inner unique pointer is non null.